lib/Connection.js

```diff
@@ -123,10 +124,34 @@ class Connection {
       callback(new RequestAbortedError(), null)
     }
 
+    const onSocket = socket => {
+      /* istanbul ignore else */
+      if (!socket.isSessionReused()) {
```

Comment:
__question:__ just for my own education, if TLS session is re-used, NodeJS doesn't issue `secureConnect`?

Reply:
It might be, but this check avoids the allocation of yet another event emitter with its own closure :)